### PR TITLE
docs: Key manager flow diagrams

### PR DESCRIPTION
**Pull request by matevz**
This PR adds three diagrams to clarify to our audience how keys are exchanged and enclaves are integrated in our stack to achieve confidentiality:
- simplified Client <-> KM <-> Compute node DAG (shown on Oasis Network and Sapphire Guide pages)
- detailed confidential contract transaction flow on Sapphire sequence diagram (Sapphire Guide page, toggable)
- detailed confidential contract call flow on Sapphire sequence diagram (Sapphire Guide page, toggable)

PR also:
- adds mmdc dependency for compiling mermaid diagrams offline into .svg 
- adds new `yarn diagrams` rule which does the above
- adds checking that .svgs are in sync with .mmd sources in CI
- describes how to deal with diagrams to README.md
- revamps Create dApp page

Preview: https://deploy-preview-310--trusting-archimedes-14c863.netlify.app/dapp/sapphire/guide#writing-secure-dapps

### CODE REVIEW

File: docs/dapp/diagrams/c10l-smart-contract-tx.mmd [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        Web3->>+ClientN: Get ephemeral<br/>public key R
+        ClientN->>+KM: Get ephemeral<br/>public key R
+        KM->>KM: Derive x25519 keypair R,R'<br/>from Master key ||<br/>Epoch || ParaTime ID
+        KM->>-ClientN: Ephemeral public key R
+        Note right of ClientN: R is cached<br/>until the end<br/>of epoch
+        ClientN->>-Web3: Ephemeral<br/>public key R
+        Web3->>-Client: Ephemeral<br/>public key R
+
+        Client->>Client: Generate x5519<br/>keypair C,C'
+        Client->>Client: Encrypt transaction body<br/>with Deoxys-II envelope<br/>using C',R
```
Comment (kostko) [inline docs/dapp/diagrams/c10l-smart-contract-tx.mmd]: This needs to be split into two steps:
* Derive shared secret `s := X25519(C', R)`
* Encrypt call data with Deoxys-II using shared secret s.

Also it is not the entire transaction body that is encrypted but just the call data.

File: docs/dapp/diagrams/c10l-smart-contract-tx.mmd [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    Web3->>+ClientN: eth_sendRawTx
+    ClientN->>ClientN: Validate<br/>transaction
+    ClientN->>Compute: eth_sendRawTx
+
+    opt Encrypted Transaction
+        Compute->>+KM: Get ephemeral<br/>private key R'
+        KM->>KM: Check runtime<br/>policy for caller
+        KM->>KM: Derive x25519 keypair R,R'<br/>from Master key ||<br/>Epoch || ParaTime ID
+        KM->>-Compute: Ephemeral<br/>private key R'
+        Compute->>Compute: Decrypt Deoxys-II<br/>envelope using C,R'
```
Comment (kostko) [inline docs/dapp/diagrams/c10l-smart-contract-tx.mmd]: Similar to above but with C, R'.

File: docs/dapp/diagrams/c10l-smart-contract-tx.mmd [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    participant Compute as Oasis Node<br/>ParaTime compute<br/>🔒
+    participant KM as Oasis Node<br/>Key manager<br/>🔒
+
+    Client->>Client: Create transaction<br/>body
+
+    opt Encrypt Transaction
+        Client->>+Web3: Get ephemeral<br/>public key R
+        Web3->>+ClientN: Get ephemeral<br/>public key R
+        ClientN->>+KM: Get ephemeral<br/>public key R
+        KM->>KM: Derive x25519 keypair R,R'<br/>from Master key ||<br/>Epoch || ParaTime ID
```
Comment (kostko) [inline docs/dapp/diagrams/c10l-smart-contract-tx.mmd]: Probably no need to specify the concatenation (as it is not just simply concatenated), just say ephemeral per-epoch per-runtime key.

File: README.md [hunk 4]
```diff
@@ -177,6 +177,25 @@ number in the comment which added this redirection for future reference, if
 major rewrite is to happen and the developers would need more context around
 the redirection.
 
+### Diagrams
+
+There are three kinds of image assets currently used in the docs.
+
+1. Screenshots, photos, non-technical figures go into `images/` folder of the
+   respected top-level chapter.
+2. [Mermaid][https://mermaid-js.github.io] diagrams (preferred tool for
```
Comment (lukaw3d) [inline README.md]: Broken link; + diagrams.net too

File: docs/dapp/emerald/README.mdx [hunk 5]
```diff
@@ -1,3 +1,7 @@
+---
+description: ABC
```
Comment (lukaw3d) [inline docs/dapp/emerald/README.mdx]: TODO?

**Review by lukaw3d — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (matevz) [reply README.md]: Too much markdown in the last days ;)

File: docs/dapp/emerald/README.mdx (hunk 5 above)
Comment (matevz) [reply docs/dapp/emerald/README.mdx]: Oops, just some debugging leftover (is category description like emerald shown when used as doccard => no).

**Pull request merged**